### DISCUSSION TITLE
detail: 'Authentication credentials were not provided.

### DISCUSSION THREAD

**Discussion by buzzlightyear2k** [netbox-community/netbox#10008] [Q&A]
Im trying to call REST API from python code and encounter this error.Can someone help me ? 
![image](https://user-images.githubusercontent.com/86591649/184567980-54bfa499-0b86-4d00-aeaf-9eea90d580bb.png)
My code : 
![image](https://user-images.githubusercontent.com/86591649/184568008-c3c41537-8d85-4fbc-8170-5357a4fbe392.png)

**Comment by candlerb** [comment]
1. Please don't post graphical screenshots. They are very hard to read, and impossible to use (e.g. to reproduce the issue, to highlight errors etc).  Just copy-paste plain text: put three backticks (` ``` `) on a line of their own before and after, to format them properly as a code block in markdown.
2. Unless you've disabled authentication, you'll need to provide an `Authorization: Token XXXX` header, see examples [here](https://docs.netbox.dev/en/stable/rest-api/authentication/).
3. Netbox v3.2 has no `/api/dcim/load-balancer/` endpoint, and AFAICS the 3.3 beta doesn't either, so presumably you are using some plugin, or a private fork of Netbox. Try accessing the core Netbox API first, e.g. `/api/dcim/devices/`.  If your problem is *only* with the plugin endpoint or fork, then discuss this with the project's maintainer.

**Reply by fesenkokd** [reply]
Hello, I have same issue. I'm using Postman as client.
URL is  https://172.17.232.65/api/ipam/vlans/  Content-Type==application/json   Authorization ==<generated Token ID>
Response - {
    "detail": "Authentication credentials were not provided."
}
In nginx log:
[18/Aug/2022:09:07:48 +0000] "GET /api/ipam/vlans/ HTTP/1.1" 403 58 "-" "PostmanRuntime/7.29.2"
Using nginx,  Netbox v3.2
I saw such cases with Apache2, but not nginx.

**Reply by candlerb** [reply]
The error tells you exactly what's wrong: you didn't provide any authentication credentials.  You need to add header

```
Authorization: Token XXXXXXXX
```

to your request (where XXXXXXXX is some token you've already generated via the web UI).  And if you're sure you've added that header, then it must be being stripped somewhere, e.g. by your proxy.

You can use tcpdump on your Netbox server to see the HTTP requests after they've been passed through the proxy:

```
tcpdump -i lo -nn -s0 -A tcp port 8001
```

**Reply by fesenkokd** [reply]
@candlerb Thank you for reply, seems issue with WSGI:
.. ... .GET /api/ipam/vlans/ HTTP/1.0
X-Forwarded-Host: 172.17.232.65
X-Real-IP: ::ffff:172.17.10.16
X-Forwarded-Proto: https
Host: 127.0.0.1:8001
Connection: close
Content-Type: application/json
Authorization: 51bc38ec59787f9ba14b985b9f9379eb4e4071c3
User-Agent: PostmanRuntime/7.29.2
Accept: */*
Postman-Token: ff1527a9-9d91-4983-b6fa-84e8f502cd90
Accept-Encoding: gzip, deflate, br


09:58:20.111615 IP 127.0.0.1.8001 > 127.0.0.1.55836: Flags [.], ack 392, win 509, options [nop,nop,TS val 2935234758 ecr 2935234758], length 0
E..4..@.@............A.....]...1.....(.....
.. ... .
09:58:20.115255 IP 127.0.0.1.8001 > 127.0.0.1.55836: Flags [P.], seq 1:387, ack 392, win 512, options [nop,nop,TS val 2935234761 ecr 2935234758], length 386
E.....@.@............A.....]...1...........
.. ... .HTTP/1.0 403 Forbidden
Server: gunicorn
Date: Thu, 18 Aug 2022 09:58:20 GMT
Connection: close
Content-Type: application/json
Vary: Accept, Cookie, Origin
Allow: GET, POST, PUT, PATCH, DELETE, HEAD, OPTIONS
API-Version: 3.2
X-Content-Type-Options: nosniff
Referrer-Policy: same-origin
Cross-Origin-Opener-Policy: same-origin
X-Frame-Options: SAMEORIGIN
Content-Length: 58

**Reply by fesenkokd** [reply]
Find this
Aug 18 09:53:48 srv-netbox-test01 gunicorn[2126]: Forbidden: /api/ipam/vlans/
Aug 18 09:54:42 srv-netbox-test01 gunicorn[2124]: Forbidden: /api/ipam/vlans/
Aug 18 09:58:20 srv-netbox-test01 gunicorn[2124]: Forbidden: /api/ipam/vlans/
Aug 18 11:50:49 srv-netbox-test01 gunicorn[2124]: Forbidden: /api/ipam/vlans/

**Reply by candlerb** [reply]
That will be because you're either not providing the Authorization header, or your proxy is not passing it through. There's nothing Netbox can do about that.

Are you *sure* the problem is not with Postman?  I have a vague recollection of coming across a problem where Postman was dropping Authorization headers, but I can't remember the details and I can't find it by Googling.  All I found was:
https://community.postman.com/t/request-headers-are-not-getting-sent/17440

If you can make it work with curl, then it's almost certainly a Postman problem.  Try this:

```
curl -k -H "Authorization: Token XXXXXXXX" https://172.17.232.65/api/ipam/vlans/ 
```

**Reply by fesenkokd** [reply]
@candlerb you absolutly right, my fault in Postman config for Auth. I choose wrong Auth for workplace, now it's works properly. Thank you!

**Reply by munazza-shahzad** [reply]
Which auth did you choose, or change in postman? I am encountering the same issue.

**Reply by rgshenoy** [reply]
My problem was that I was using Bearer token in Postman and it would add Auth header as "Authorization: **Bearer** Token  xxxxxxx" whereas it should have been just "Authorization: Token xxxxxxx". I manually added Authorization header instead of using Authorization tab in postman and it worked fine.
